src/core/main.c

```diff
@@ -668,26 +698,34 @@ static int parse_config_file(void) {
                 {}
         };
 
-        const char *fn, *conf_dirs_nulstr;
+        _cleanup_strv_free_ char **_free_files = NULL, **_free_dirs = NULL;
 
-        fn = arg_system ?
-                PKGSYSCONFDIR "/system.conf" :
-                PKGSYSCONFDIR "/user.conf";
+        const char *const *files, *const *dirs, *suffix;
+        int r;
 
-        conf_dirs_nulstr = arg_system ?
-                CONF_PATHS_NULSTR("systemd/system.conf.d") :
-                CONF_PATHS_NULSTR("systemd/user.conf.d");
+        if (arg_system) {
+                files = STRV_MAKE_CONST(PKGSYSCONFDIR "/system.conf");
+                dirs = (const char* const*) CONF_PATHS_STRV("systemd");
+                suffix = "system.conf.d";
+        } else {
+                r = manager_find_user_config_paths(&_free_files, &_free_dirs);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to determine config file paths: %m");
+                files = (const char* const*) _free_files;
+                dirs = (const char* const*) _free_dirs;
```

Comment:
this is ugly!

let's just accept that strv arrays are const-less, even if it sucks...